validator/sawtooth_validator/concurrent/atomic.py

```diff
@@ -128,3 +128,32 @@ def get(self, key, default):
                 return self._dict[key].copy()
             except KeyError:
                 return default
+
+
+class ConsensusRegistry:
+    """A thread-safe construct that stores the connection_id, name, and version
+    of the currently registered consensus engine."""
+
+    def __init__(self):
+        self._info = None
+        self._lock = RLock()
+
+    def __bool__(self):
+        with self._lock:
+            return self._info is not None
+
+    def register_engine(self, connection_id, name, version):
+        with self._lock:
+            self._info = {
+                'connection_id': connection_id,
+                'name': name,
+                'version': version
+            }
```

Comment:
This should be a namedtuple here to lock in the fields and especially since you are returning a reference to a thing that is guarded by a lock out from under the lock. If you use a dictionary, after calling `get_engine_info()`, you could modify the dictionary without acquiring the lock. namedtuples are a lot like dictionaries, but they are immutable so it doesn’t matter if you return it out from under the lock and you can't add/remove fields.